isis/src/base/objs/CSMCamera/CSMCamera.cpp

```diff
@@ -167,12 +148,56 @@ namespace Isis {
     return PixelResolution();
   }
 
-  // Return the target
-  Target *CSMCamera::target() const {
-    return m_target;
+
+  double CSMCamera::parentLine() {
+    return p_alphaCube->AlphaLine(Line());
+  }
+
+  double CSMCamera::parentSample() {
+    return p_alphaCube->AlphaSample(Sample());      
+  }
+
+
+  // Broke sensorPosition call out to separate function in preparation for in the future separating the corresponding
+  // call out of Spice::subSpacecraft to decrease repeated code and just override this function.
+  std::vector<double> CSMCamera::sensorPositionBodyFixed(){
+    return sensorPositionBodyFixed(parentLine(), parentSample());
+  }
+  
+
+  // stateless version
+  std::vector<double> CSMCamera::sensorPositionBodyFixed(double line, double sample){
+    csm::ImageCoord imagePt(line, sample);
```

Comment:
We need to account for the differences in origin for the ISIS and CSM pixels. We need to subtract 0.5 from the line and sample everytime we pass a pixel location to CSM. Maybe a small helper function that takes line, sample in ISIS, subtracts 0.5 from each, and then return a csm::ImageCoord.